src/Kokkos_ArithTraits.hpp

```diff
@@ -849,7 +849,7 @@ template<class RealFloatType>
 class ArithTraits<std::complex<RealFloatType> > {
 public:
   //! Kokkos internally replaces std::complex with Kokkos::complex.
-  typedef ::Kokkos::complex<RealFloatType> val_type;
+  typedef std::complex<RealFloatType> val_type;
```

Comment:
@mhoemmen was that intentional?  I am confused now because they are later specializations for `Kokkos::complex` in this file.

Reply:
Don't do this.  Read the comment :)  It's intentional that `val_type` is `::Kokkos::complex<RealFloatType>`.  The point is that `val_type` works with CUDA and works as the result of Kokkos reductions.  It needs to be bitwise equivalent (*) to ArithTraits' template parameter.  There are specializations for both std::complex and Kokkos::complex, because users need to make Views of Kokkos::complex in order for their code to work on CUDA.  std::complex's methods lack the `__device__ __host__` qualifiers.

(*) std::complex technically could use a different representation than (real, imag).  However, we haven't had any problems assuming that this is its representation.

Reply:
Fair enough.  I reverted the change.